Exiled.Events/Patches/Events/Player/ChangingRole.cs

```diff
@@ -76,11 +76,14 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                 new(OpCodes.Newobj, GetDeclaredConstructors(typeof(ChangingRoleEventArgs))[0]),
                 new(OpCodes.Dup),
                 new(OpCodes.Dup),
+                new(OpCodes.Dup),
                 new(OpCodes.Stloc_S, ev.LocalIndex),
 
                 // Handlers.Player.OnChangingRole(ev)
                 new(OpCodes.Call, Method(typeof(Player), nameof(Player.OnChangingRole))),
 
+                new(OpCodes.Callvirt, Method(typeof(ChangingRoleEventArgs), nameof(ChangingRoleEventArgs.ConsolidatePlayerInformation))),
+
                 // if (!ev.IsAllowed)
                 //    return;
                 new(OpCodes.Callvirt, PropertyGetter(typeof(ChangingRoleEventArgs), nameof(ChangingRoleEventArgs.IsAllowed))),
```

Comment:
🙂 now it will be broke when `ev.IsAllowed = false`

Reply:
If the event is false their role does not change

Reply:
Yes, I overreacted, but logically it would be correct to put it after the IsAllowed check(You've already done it.)